### PR TITLE
Added discussion admin role to coach and prepared discussion forum for CCX

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
### Background
This PR resolves https://github.com/mitocw/edx-platform/issues/136

**LMS Updates:** 
- Enabled discussion forum and (inline) discussion XBlock for CCX.
- Prepared discussion forum for CCX 
- Assigned coach of CCX admin role on discussion form.
- Added discussion enable/disable flag ``enable_discussion`` in  **CustomCourseForEdX**
- Create migration for ``enable_discussion`` flag
- Add UI to enable or disable Discussion forum.
- [ ] Added performance tests for inline discussion (loading views).

@pdpinch 

screen shot 2016-01-28 at 3 45 19 pm

### CODE REVIEW

File: lms/djangoapps/django_comment_client/utils.py [hunk 1]
```diff
@@ -76,6 +76,19 @@ def get_role_ids(course_id):
     return dict([(role.name, list(role.users.values_list('id', flat=True))) for role in roles])
 
 
+def assign_discussion_role(course_key, user, role=FORUM_ROLE_STUDENT):
+    """
+    Assigns discussion role to user, default role is 'FORUM_ROLE_STUDENT'
```
Comment (pdpinch) [inline lms/djangoapps/django_comment_client/utils.py]: How is it that this function doesn't already exist?

**Comment by pdpinch** [comment]
I'm not sure what the 2nd bullet means, "Prepared discussion forum for CCX " 

I don't see anything for the last bullet -- but that's OK because I think that should be a separate PR. I'm already wondering if we should break this one down into one for enabling and one for assigning.

**Comment by amir-qayyum-khan** [comment]
@pdpinch this in in progress PR, I have a question. I want to save (can be boolean) a flag for discussion form enable/disable where should i please it.

- In course (just like ccx connector flag or enable ccx flag)
**OR**
- In CustomCourseForEdX mode  and create migration for existing CCXs

??

**Comment by pdpinch** [comment]
Add the discussion flag to the CustomCourseForEdX model. It needs to be scoped to the CCX, not the master course. 

Also, edX prefers we extend that model. Putting settings in the course can be kind of expensive.

**Comment by amir-qayyum-khan** [comment]
ok, thanks

**Pull request closed without merging**